tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc

```diff
@@ -1378,22 +1394,34 @@ LogicalResult ConvertTFFusedBatchNormOp::matchAndRewrite(
   // op5 = mul(op4, bscale)
   // op6 = add(op5, boffset)
 
-  RankedTensorType mean_type =
-      tf_batchnorm_op.getMean().getType().dyn_cast<RankedTensorType>();
+  Value mean = tf_batchnorm_op.getMean();
+  Value variance = tf_batchnorm_op.getVariance();
+  Value x = tf_batchnorm_op.getX();
+  Value scale = tf_batchnorm_op.getScale();
+  Value offset = tf_batchnorm_op.getOffset();
+
+  RankedTensorType mean_type = mean.getType().dyn_cast<RankedTensorType>();
   RankedTensorType variance_type =
-      tf_batchnorm_op.getVariance().getType().dyn_cast<RankedTensorType>();
-  if (!variance_type || !mean_type) return failure();
+      variance.getType().dyn_cast<RankedTensorType>();
+  RankedTensorType x_type = x.getType().dyn_cast<RankedTensorType>();
```

Comment:
New MLiR style is use dyn_cast<T>(...) style (see deprecation page on mlir.dev).

Reply:
will fix all dyn_cast to new style in a separate PR

Reply:
fixed to use dyn_cast(...) throughout